updatesnap/unittests.py

```diff
@@ -423,6 +426,33 @@ def test_invalid_version_variation_and_beta_release(self):
                 part['part_name'], part['version'], part['entry_format'], False)
             assert version is None
 
+    def test_snap_version_automation(self):
+        """ tests if snap version automation working correctly"""
+        data = self._base_load_test_file("test_snap_version_automation.yaml")
+        yaml_obj = ManageYAML(data)
+        snap, _, _, _ = self._load_test_file("test_snap_version_automation.yaml",
+                                             None)
+        args = Namespace(
+            version_schema=r'^debian/(\d+\.\d+\.\d+)',
+        )
+        logging.basicConfig(level=logging.ERROR)
+        test = is_version_update(snap, yaml_obj, args)
+        assert test is not None
```

Comment:
I would also check that the new version number is correct.

Reply:
I'm not sure we can reliably check this because versions can change, and I can't keep track of every change to verify if it's correct.

Reply:
Can't you write a mock function that uses only data from disk, to ensure reproducibility?

Reply:
The "is_version_update" function relies on data from the snap store (it fetches data using snap API), which means it requires a snap to be listed there. It's important to remember that the version of a snap on the snap store might change unexpectedly.

Reply:
I see that you just call a method from "snap" (process_metadata()) and another from manager_yaml (get_part_metadata()). You can mock both in the objects passed to that function.

Reply:
'is_version_update' function internally uses 'process_snap_version_data' which uses snap API to fetch data

Reply:
You can replace that function in the module with a mockup too.  just create the function and assign it to `module_name.function_name`. Of course, I would move just the part that gets the json from the server into a function, and mock only that function instead, returning a json with the desired test data. And also would first save the original function, to restore it after the test.

Reply:
@sergio-costas, I have updated `unittests` according to above discussion, do check them